docs/glossary.md

```diff
@@ -0,0 +1,83 @@
+---
+layout: default
+title: lakeFS Glossary
+description: Glossary of all terms related to lakeFS technical internals and the architecture
+nav_order: 52
+has_children: false
+---
+
+# Glossary
+This page has definition and explanations of all terms related to lakeFS technical internals and the architecture.
+
+{% include toc.html %}
+
+### Auditing
+Data auditing is data assessment to ensure its accuracy, security, and efficacy for specific usage. It also involves assessing data quality through its lifecycle and understanding the impact of poor quality data on the organization's performance and revenue. Ensuring data reproducibility, auditability, and governance is one of the key concerns of data engineers today. lakeFS commit history helps the data teams to keep track of all changes to the data, and thus support data auditing.
+
+### Branch
+
+Branches are similar in concept to Git branches.
+When creating a new branch, we are actually creating a consistent snapshot of the entire repository, which is isolated from other branches and their changes.
+
+A branch is a mutable pointer to a commit and uncommitted changes in its staging area (i.e., mutable storage where users can create, update, and delete objects). When a user creates a commit from a branch, all the files from the staging area will be merged into the contents of the current branch, generating a new set of objects. The pointer is updated to reference the new set of objects. The new branch tip is set to the latest commit and the previous branch tip serves as the commit's parent. 
+
+Just like in git, a branch spans a repository. Learn more about the [lakeFS branching model](/understand/branching-model.md).
```

Comment:
Links should be relative, otherwise they will break under versioning. Give as many `..`s as needed, then go back down.